src/routes/[topTierPage]/+page.server.ts

```diff
@@ -0,0 +1,48 @@
+import gql from "graphql-tag";
+import { print as printQuery } from "graphql";
+import { error } from "@sveltejs/kit";
+import { CONTENTFUL_SPACE_ID, CONTENTFUL_DELIVERY_API_TOKEN } from "$env/static/private";
+import getContentfulClient from "$lib/services/contentful";
+
+import type { TopTierCollectionQuery } from "./$queries.generated";
+
+const query = gql`
+  query TopTierCollection {
+    topTierCollection(limit: 10) {
+      items {
+        pageMetadata {
+          ... on PageMetadata {
+            sys {
+              id
+            }
+            slug
+          }
+        }
+      }
+    }
+  }
+`;
+
+export const load = async ({ parent, params }) => {
+  const { pageMetadataMap } = await parent();
```

Comment:
The issue with this pattern is that we're effectively calling `loadPageMetadataMap` every time we load (or preload) a page, which is pretty terrible. I'd be happier with a solution where this only gets called once per session on the first page load, but I think ideally we could store this map somewhere for use by all sessions and only update it when Contentful gets updated. 

What do y'all think is the best way to make a small improvement here? I'm thinking about trying to use either `store` or `setContext` / `getContext` but I'm not sure which one would be a better fit here.

Reply:
I think that the Context API is the way to go here, primarily because:

1. `pageMetadataMap` is not reactive, so we don't need to subscribe to it like a store value.
2. since `pageMetadataMap` is initialized from the root, it should be accessible to all other pages via context, since they'll all be children.

Then we could just store this in either localStorage, or would this perhaps be an appropriate use case for service workers? I haven't worked with them before, so I'm wary of a cache not properly updating and leading to stale data, especially with something like route generation, but in either case, _some_ sort of persistence mechanism should work in tandem with `setContext`/`getContext`

Reply:
Actually...since SvelteKit already passes propagates data from parent `layout`s down to their children, I don't think we even need to worry about using Context. Simply caching the call should be enough